### PR TITLE
Normalize shopware recipe and add common.php

### PR DESCRIPTION
**Pull request by Schrank**
- [ ] Bug fix #…?
- [ ] New feature?
- [ ] BC breaks?
- [ ] Deprecations?
- [ ] Tests added?
- [X] Changelog updated?

### CODE REVIEW

File: docs/recipe/shopware.md [hunk 1]
```diff
@@ -34,26 +36,36 @@
 ### repository
 [Source](https://github.com/deployphp/deployer/search?q=%22repository%22+in%3Afile+language%3Aphp+path%3Arecipe+filename%3Ashopware.php)
 
+* Overrides [`repository`](/docs/recipe/common.md#repository) from `recipe/common.php`
```
Comment (Schrank) [inline docs/recipe/shopware.md:39]: @peterjaap maybe remove set('repository')?

Comment (peterjaap) [reply docs/recipe/shopware.md:39]: Yeah possible, but I'd rather leave it in to remind people of what to do (for those less familiar with Deployer)

**Comment by peterjaap** [comment]
Why is this necessary? Isn't common automatically loaded by Deployer?

**Comment by Schrank** [comment]
Good question, need to check.

I assumed no and this recipe is the base for `dep init` For shopware and therefore should require it. Will check and if deployer takes care of it remove it from all other recipes 😂

**Comment by peterjaap** [comment]
I check my installs and we usually do this;

```
<?php
namespace Deployer;

require 'vendor/deployer/deployer/recipe/common.php';
require 'vendor/deployer/deployer/recipe/shopware.php';
```

So adding it through `require_once` should be backwards compatible with that.

**Comment by antonmedv** [comment]
So, should we merge it?

**Comment by Schrank** [comment]
I’ll have a look and will take care of it :/)

**Comment by Schrank** [comment]
common.php is needed and not auto-included by deployer.

File: recipe/shopware.php [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 use MJS\TopSort\Implementations\FixedArraySort;
```
Comment (antonmedv) [inline recipe/shopware.php:5]: Maybe use independent implementation? And recipe can work without autoloader.

Comment (Schrank) [reply recipe/shopware.php:5]: Great idea. I never implemented a topological sorting :) And I think as long as no one is complaining it has low priority :)

**Pull request merged**